typescript/agentkit/src/action-providers/compound/compoundActionProvider.test.ts

```diff
@@ -0,0 +1,389 @@
+import { parseEther, parseUnits, encodeFunctionData } from "viem";
+import { EvmWalletProvider } from "../../wallet-providers/evmWalletProvider";
+import { CompoundActionProvider } from "./compoundActionProvider";
+import { COMET_ABI, COMET_ADDRESSES, ASSET_ADDRESSES } from "./constants";
+import { Network } from "../../network";
+import { approve } from "../../utils";
+
+jest.mock("../../utils");
+const mockApprove = approve as jest.MockedFunction<typeof approve>;
+
+describe("Compound Action Provider", () => {
+  const actionProvider = new CompoundActionProvider();
+  let mockWallet: jest.Mocked<EvmWalletProvider>;
+
+  const MOCK_NETWORK: Network = { protocolFamily: "evm", networkId: "base-mainnet" };
+  const MOCK_COMET_ADDRESS = COMET_ADDRESSES["base-mainnet"];
+  const MOCK_WETH_ADDRESS = ASSET_ADDRESSES["base-mainnet"]["weth"];
+  const MOCK_USDC_ADDRESS = ASSET_ADDRESSES["base-mainnet"]["usdc"];
+  const MOCK_TX_HASH = "0xtesttxhash";
+  const MOCK_RECEIPT = { status: 1, blockNumber: 123456 };
+  const MOCK_WALLET_BALANCE = BigInt("10000000000000000000"); // 10 tokens in wei
+  const MOCK_TOKEN_SYMBOL = "WETH";
+
+  beforeEach(() => {
+    mockWallet = {
+      getAddress: jest.fn().mockResolvedValue("0xMockAddress"),
+      getNetwork: jest.fn().mockReturnValue(MOCK_NETWORK),
+      sendTransaction: jest.fn().mockResolvedValue(MOCK_TX_HASH as `0x${string}`),
```

Comment:
Please update the usages of `0x${string}` in this file to `Hex` or `Address` from viem
```suggestion
      sendTransaction: jest.fn().mockResolvedValue(MOCK_TX_HASH as Hex),
```